lib/src/components/NxList/types.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { HTMLAttributes, LiHTMLAttributes, WeakValidationMap, ReactElement,
+  ReactNode, ButtonHTMLAttributes, AnchorHTMLAttributes } from 'react';
+import PropTypes from 'prop-types';
+
+export type NxListProps = HTMLAttributes<HTMLUListElement> & {
+  bulleted?: boolean;
+  children?: ReactElement | ReactElement[] | null | ReactNode;
+};
+
+export const nxListPropTypes: WeakValidationMap<NxListProps> = {
+  bulleted: PropTypes.bool,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.element.isRequired).isRequired,
+    PropTypes.element.isRequired
+  ])
+};
+
+export type NxListItemProps = LiHTMLAttributes<HTMLLIElement> & {
+  selected?: boolean;
+};
```

Comment:
Interfaces are supposed to be slightly better performing
```suggestion
export interface NxListItemProps extends LiHTMLAttributes<HTMLLIElement>  {
  selected?: boolean;
}
```